pkg/plugins/common/kustomize/v1/scaffolds/init.go

```diff
@@ -62,23 +62,44 @@ func (s *initScaffolder) Scaffold() error {
 		machinery.WithConfig(s.config),
 	)
 
-	return scaffold.Execute(
-		&rbac.Kustomization{},
-		&rbac.AuthProxyRole{},
-		&rbac.AuthProxyRoleBinding{},
-		&rbac.AuthProxyService{},
-		&rbac.AuthProxyClientRole{},
-		&rbac.RoleBinding{},
-		&rbac.LeaderElectionRole{},
-		&rbac.LeaderElectionRoleBinding{},
-		&rbac.ServiceAccount{},
-		&manager.Kustomization{},
-		&manager.Config{Image: imageName},
-		&manager.ControllerManagerConfig{},
-		&kdefault.Kustomization{},
-		&kdefault.ManagerAuthProxyPatch{},
-		&kdefault.ManagerConfigPatch{},
-		&prometheus.Kustomization{},
-		&prometheus.Monitor{},
-	)
+	if s.config.IsComponentConfig() {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&manager.ControllerManagerConfig{},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+	} else {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+	}
```

Comment:
From what I can tell here, the only difference in the scaffolds that needs to happen is that the template `&manager.ControllerManagerConfig{}` isn't included by default.

To reduce the repetition and increase readability I think we could do something like:

```go
templates := []machinery.Template{
  &rbac.Kustomization{},
  &rbac.AuthProxyRole{},
  &rbac.AuthProxyRoleBinding{},
  &rbac.AuthProxyService{},
  &rbac.AuthProxyClientRole{},
  &rbac.RoleBinding{},
  &rbac.LeaderElectionRole{},
  &rbac.LeaderElectionRoleBinding{},
  &rbac.ServiceAccount{},
  &manager.Kustomization{},
  &manager.Config{Image: imageName},
  &kdefault.Kustomization{},
  &kdefault.ManagerAuthProxyPatch{},
  &kdefault.ManagerConfigPatch{},
  &prometheus.Kustomization{},
  &prometheus.Monitor{},
}

if s.config.IsComponentConfig() {
  templates = append(templates, &manager.ControllerManagerConfig{})
}

return scaffold.Execute(templates...)
```
@camilamacedo86 @laxmikantbpandhare WDYT?

Reply:
Yes, sounds perfect!

Reply:
```
templates := []machinery.Builder{
  &rbac.Kustomization{},
  &rbac.AuthProxyRole{},
  &rbac.AuthProxyRoleBinding{},
  &rbac.AuthProxyService{},
  &rbac.AuthProxyClientRole{},
  &rbac.RoleBinding{},
  &rbac.LeaderElectionRole{},
  &rbac.LeaderElectionRoleBinding{},
  &rbac.ServiceAccount{},
  &manager.Kustomization{},
  &manager.Config{Image: imageName},
  &kdefault.Kustomization{},
  &kdefault.ManagerAuthProxyPatch{},
  &kdefault.ManagerConfigPatch{},
  &prometheus.Kustomization{},
  &prometheus.Monitor{},
}

if s.config.IsComponentConfig() {
  templates = append(templates, &manager.ControllerManagerConfig{})
}

return scaffold.Execute(templates...)
```

Reply:
Updated and ready for review @everettraven